src/rust/catnip/runtime/mod.rs

```diff
@@ -174,8 +174,9 @@ impl DPDKRuntime {
         std::env::set_var("MLX5_SINGLE_THREADED", "1");
         std::env::set_var("MLX4_SINGLE_THREADED", "1");
         let eal_init_refs = eal_init_args.iter().map(|s| s.as_ptr() as *mut u8).collect::<Vec<_>>();
-        unsafe {
-            rte_eal_init(eal_init_refs.len() as i32, eal_init_refs.as_ptr() as *mut _);
+        let ret: libc::c_int = unsafe { rte_eal_init(eal_init_refs.len() as i32, eal_init_refs.as_ptr() as *mut _) };
+        if ret < 0 {
+            bail!("EAL initialization failed {:?}", ret);
```

Comment:
Is there any way we can get access to `rte_errno` here?  And print that instead of the return value?  `rte_eal_init()` is documented as returning -1 on error, and "`rte_errno` is set to a value indicating the cause for failure".

Reply:
Thanks for noting this. I've updated the fix for getting and printing the value of `rte_errno`.

Reply:
Awesome!  Thanks.